Source/Matter/ConstraintsWithMatter.impl.hpp

```diff
@@ -0,0 +1,125 @@
+/* GRTeclyn
+ * Copyright 2022 The GRTL collaboration.
+ * Please refer to LICENSE in GRTeclyn's root directory.
+ */
+
+#if !defined(CONSTRAINTSWITHMATTER_HPP_)
+#error "This file should only be included through ConstraintsWithMatter.hpp"
+#endif
+
+#ifndef CONSTRAINTSWITHMATTER_IMPL_HPP_
+#define CONSTRAINTSWITHMATTER_IMPL_HPP_
+#include "DimensionDefinitions.hpp"
+#include "GRParmParse.hpp"
+
+template <class matter_t>
+ConstraintsWithMatter<matter_t>::ConstraintsWithMatter(
+    double dx, double G_Newton, int a_c_Ham, const Interval &a_c_Moms,
+    int a_c_Ham_abs_terms /* defaulted*/,
+    const Interval &a_c_Moms_abs_terms /*defaulted*/)
+    : Constraints(dx, a_c_Ham, a_c_Moms, a_c_Ham_abs_terms, a_c_Moms_abs_terms,
+                  0.0 /*No cosmological constant*/),
+      m_G_Newton(G_Newton)
+{
+}
+
+template <class matter_t>
+template <class data_t>
+AMREX_GPU_DEVICE AMREX_FORCE_INLINE void
+ConstraintsWithMatter<matter_t>::compute(
+    int i, int j, int k, const amrex::Array4<data_t> &cst,
+    const amrex::Array4<data_t const> &state) const
```

Comment:
Will need to change the variables in the function below too...
```suggestion
    int i, int j, int k, const amrex::Array4<data_t> &out_arrays,
    const amrex::Array4<data_t const> &state_arrays) const
```